### PR TITLE
Fix CWE-476 and remove not used method in WebDav

### PR DESCRIPTION
**Pull request by Beacze**
http://cwe.mitre.org/data/definitions/476.html

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/services/validation/MetadataValidationService.java [hunk 1]
```diff
@@ -101,24 +101,29 @@ public boolean validate(Fileformat gdzfile, Prefs prefs, Process process) {
...
+                        parameter.add(logical.getType().getName());
+                        parameter.add(firstChild.getType().getName());
+                        Helper.setFehlerMeldung(Helper.getTranslation("InvalidIdentifierSame", parameter));
+                        result = false;
+                    }
 
-                result = checkIfMetadataValueNotReplaced(firstChild, identifierFirstChild, metadataLanguage);
+                    result = checkIfMetadataValueNotReplaced(firstChild, identifierFirstChild, metadataLanguage);
+                }
+                //TODO: why catch here, if this block doesn't throw any exception??
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/validation/MetadataValidationService.java]: If this try catch block is not needed, then remove it.

File: Kitodo/src/main/java/org/kitodo/services/schema/SchemaService.java [hunk 2]
```diff
@@ -494,6 +494,7 @@ private void addMissingMetadata(DocStruct inTopStruct, Prefs prefs, Process proc
         Metadata mdPPNBand = null;
         Metadata mdSorting = null;
         try {
+            //TODO: why is it created here and next not assigned?
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/schema/SchemaService.java]: Is this TODO for the `mdTitle`? If so and  `mdTitle` is not used yet, then if could be removed.

File: Kitodo/src/main/java/de/sub/goobi/helper/WebDav.java [hunk 3]
```diff
@@ -62,9 +62,14 @@ public WebDav() {
...
 
         try {
-            URI directoryName = serviceManager.getUserService().getHomeDirectory(currentUser).resolve(inVerzeichnis);
-            files = fileService.getSubUris(filter, directoryName);
-        } catch (Exception ioe) {
+            if (currentUser != null) {
+                URI directoryName = serviceManager.getUserService().getHomeDirectory(currentUser).resolve(inVerzeichnis);
+                files = fileService.getSubUris(filter, directoryName);
+            } else {
+                Helper.setFehlerMeldung("uploadFromHomeAlle abgebrochen, Fehler - no user assigned");
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/helper/WebDav.java]: Please, do not mix German and English. Please use one language or use a translation key for translatable error messages.

Comment (Beacze) [reply Kitodo/src/main/java/de/sub/goobi/helper/WebDav.java]: Is there a key for such a message in German or better whole message in English?

File: Kitodo/src/main/java/org/kitodo/services/validation/MetadataValidationService.java (hunk 1 above)
Comment (Beacze) [reply Kitodo/src/main/java/org/kitodo/services/validation/MetadataValidationService.java]: I have looked at this again and it looks this block was only for catching NPE and logging. SO I remove it and move logger to else block.

File: Kitodo/src/main/java/de/sub/goobi/helper/WebDav.java (hunk 3 above)
Comment (henning-gerhardt) [reply Kitodo/src/main/java/de/sub/goobi/helper/WebDav.java]: No, this must be created for both languages as you added this in this pull request.

**Comment by henning-gerhardt** [comment]
There is at least one merge conflict in `Kitodo/src/main/java/de/sub/goobi/metadaten/Metadaten.java`. Please solve this conflict.

**Pull request merged**